base/commands/migration/migration_stages.go

```diff
@@ -233,6 +241,46 @@ func fetchMigrationStatus(ctx context.Context, ci *hazelcast.ClientInternal, mig
 	return strings.TrimSuffix(strings.TrimPrefix(string(r.(serialization.JSON)), `"`), `"`), nil
 }
 
+func fetchOverallProgress(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (time.Duration, float32, error) {
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.remainingTime'), JSON_QUERY(this, '$.completionPercentage') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return 0, 0, err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return 0, 0, err
+	}
+	if it.HasNext() {
+		// single iteration is enough that we are reading single result for a single migration
+		row, err := it.Next()
+		if err != nil {
+			return 0, 0, err
+		}
+		remainingTime, err := row.Get(0)
+		if err != nil {
+			return 0, 0, err
+		}
+		completionPercentage, err := row.Get(1)
+		if err != nil {
+			return 0, 0, err
+		}
+		// convert it to parsable duration format
+		rtStr := fmt.Sprintf("%sms", remainingTime.(serialization.JSON).String())
+		rt, err := time.ParseDuration(rtStr)
+		if err != nil {
+			return 0, 0, err
+		}
+		cpStr := completionPercentage.(serialization.JSON).String()
+		cp, err := strconv.ParseFloat(cpStr, 32)
```

Comment:
why not 64?

Reply:
Statuser's `SetProgress` method accepts `float32` values.